Source/Core/Core/IOS/Network/IP/Top.h

```diff
@@ -100,5 +121,8 @@ class NetIPTopDevice : public Device
   IPCReply HandleICMPPingRequest(const IOCtlVRequest& request);
 
   Common::SocketContext m_socket_context;
+  Common::WorkQueueThread<AsyncTask> m_queue;
+  std::mutex m_async_lock;
```

Comment:
`m_async_reply_lock` (otherwise it's unclear whether this protects the *async* task queue or the *async* reply queue)